[clang-format][CMake] Generate formatting options docs during build (#113739)
llvm/llvm-project: clang/docs/ClangFormatStyleOptions.rst

    subdirectories. This is also possible through the command line, e.g.:
    ``--style={BasedOnStyle: InheritParentConfig, ColumnLimit: 20}``

> .. This section of the file is automatically generated by the
>   dump_format_style.py tool and should be updated manually.
> .. START_FORMAT_STYLE_OPTIONS

Reviewer: ```suggestion
.. START_FORMAT_STYLE_OPTIONS

..
  This section is automatically generated by the dump_format_style.py script and
  should not be updated manually.
```
